nowledge-mem-codex-plugin/hooks/nmem-stop-save.py

```
@@ -0,0 +1,162 @@
+    import_payload = {
+        "title": parsed.get("title"),
+        "messages": messages,
+    }
+
+    with tempfile.NamedTemporaryFile("w", suffix=".json", delete=False, encoding="utf-8") as handle:
+        json.dump(import_payload, handle, ensure_ascii=False)
+        temp_path = Path(handle.name)
+
+    try:
+        command = [nmem_bin, "--json", "t", "import", "-f", str(temp_path), "--id", thread_id, "-s", "codex"]
+        result = subprocess.run(command, capture_output=True, text=True, env=os.environ.copy())
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)